rotkehlchen/db/updates.py

```diff
@@ -154,29 +156,72 @@ def update_rpc_nodes(self, to_version: int) -> None:
+                ))
+
+            new_contracts_data = []
+            for single_contract_data in updated_contracts['contracts_data']:
+                new_contracts_data.append((
+                    single_contract_data['address'],
+                    single_contract_data['chain_id'],
+                    single_contract_data.get('name'),
+                    single_contract_data['abi'],
+                    single_contract_data.get('deployed_block'),
+                ))
+
+            try:
+                with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                    write_cursor.executemany(
+                        'INSERT OR REPLACE INTO contract_abi(id, value, name) VALUES(?, ?, ?)',
+                        new_abi_data,
+                    )
+                    write_cursor.executemany(
+                        'INSERT INTO contract_data(address, chain_id, name, abi, deployed_block) VALUES(?, ?, ?, ?, ?)',  # noqa: E501
+                        new_contracts_data,
+                    )
+            except IntegrityError as e:
+                log.error(f'Got error during updating contracts info: {str(e)}')
+                return
+
+            with self.user_db.conn.write_ctx() as write_cursor:
+                write_cursor.execute(
+                    'INSERT OR REPLACE INTO settings(name, value) VALUES (?, ?)',
+                    (UpdateType.CONTRACTS.serialize(), version),
+                )
+
     def check_for_updates(self) -> None:
         """Retrieve the information about the latest available update"""
         remote_information = self._get_remote_info_json()
 
-        # Get latest applied versions
-        with self.user_db.conn.read_ctx() as cursor:
-            local_spam_assets_version = self._check_for_last_version(
-                cursor=cursor,
-                update_type=UpdateType.SPAM_ASSETS,
-            )
-            local_rpc_nodes_version = self._check_for_last_version(
-                cursor=cursor,
-                update_type=UpdateType.RPC_NODES,
-            )
-
-        # Update all remote data
-        latest_spam_assets_version = remote_information[UpdateType.SPAM_ASSETS.value]['latest']
-        if local_spam_assets_version < latest_spam_assets_version:
-            self.update_spam_assets(local_spam_assets_version, latest_spam_assets_version)
+        for update_type in UpdateType:
+            # Get latest applied version
+            with self.user_db.conn.read_ctx() as cursor:
+                local_version = self._check_for_last_version(
+                    cursor=cursor,
+                    update_type=update_type,
+                )
 
-        latest_rpc_nodes_version = remote_information[UpdateType.RPC_NODES.value]['latest']
-        if local_rpc_nodes_version < latest_rpc_nodes_version:
-            self.update_rpc_nodes(latest_rpc_nodes_version)
+            # Update all remote data
+            latest_version = remote_information[update_type.value]['latest']
```

Comment:
In this case you also need to add a key in the dict https://github.com/rotki/data/blob/develop/updates/info.json if we don't have contracts to add maybe you can do a `.get(` instead and if is None just not execute the logic